### PR TITLE
Add national society indicators

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- https://github.com/toggle-corp/ifrc-go-meta/issues/354

## Changes
- Add national society indicators

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    const { countryId } = useOutletContext<CountryOutletContext>();
+
+    const {
+        response: indicatorResponse,
+    } = useRequest({
+        url: '/api/v2/country/{id}/databank/',
+        skip: isNotDefined(countryId),
+        pathVariables: {
+            id: Number(countryId),
+        },
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]:
```suggestion
        pathVariables: {
            id: isDefined(countryId) ? Number(countryId) : undefined,
        },
```

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import styles from './styles.module.css';
+
+function NationalSocietyIndicators() {
+    const strings = useTranslation(i18n);
+
+    const { countryId } = useOutletContext<CountryOutletContext>();
+
+    const {
+        response: indicatorResponse,
+    } = useRequest({
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: we should add `pending` as well and update UI accordingly

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                />
+                <TextOutput
+                    label={strings.nationalSocietyExpendituresLabel}
+                    value={indicatorResponse?.expenditures}
+                    valueType="number"
+                    strongValue
+                />
+                <TextOutput
+                    label={strings.nationalSocietyYouthYearLabel}
+                    value={indicatorResponse?.people_age_6_12}
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: Let's sum age from 6 to 29

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        indicatorResponse?.female_volunteer_age_13_17,
+        indicatorResponse?.female_volunteer_age_18_29,
+        indicatorResponse?.female_volunteer_age_18_49,
+        indicatorResponse?.female_volunteer_age_30_39,
+        indicatorResponse?.female_volunteer_age_40_49,
+        indicatorResponse?.female_volunteer_age_50_59,
+        indicatorResponse?.female_volunteer_age_60_69,
+        indicatorResponse?.female_volunteer_age_70_79,
+        indicatorResponse?.female_volunteer_age_80,
+    ]);
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: Don't we have the total value for each from the server?

Comment (barshathakuri) [reply src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: No, we don't have total value in the server.

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    const staffValue = sumSafe([
+        indicatorResponse?.female_staff_total,
+        indicatorResponse?.male_staff_total,
+    ]);
+
+    const volunteersValue = sumSafe([
+        indicatorResponse?.male_volunteer_total,
+        indicatorResponse?.female_volunteer_total,
+    ]);
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: what about `other` staff?

Comment (samshara) [reply src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: we should use `KPI_PStaff_Tot, KPI_PeopleVol_Tot` indicators

Comment (k9845) [reply src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: @samshara lets note that we are including 3rd gender population.

File: src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    const youthValue = sumSafe([
+        indicatorResponse?.male_volunteer_age_6_12,
+        indicatorResponse?.male_volunteer_age_13_17,
+        indicatorResponse?.male_volunteer_age_18_29,
+        indicatorResponse?.female_volunteer_age_6_12,
+        indicatorResponse?.female_volunteer_age_13_17,
+        indicatorResponse?.female_volunteer_age_18_29,
+        indicatorResponse?.female_staff_age_18_29,
+        indicatorResponse?.male_staff_age_18_29,
+    ]);
```
Comment (samshara) [inline src/views/CountryNsOverviewContextAndStructure/NationalSocietyIndicators/index.tsx]: As per previous discussions, we should use the total value.
cc: @k9845
Indicators to use
```
KPI_PeopleVol_Tot_age_6_12
KPI_PeopleVol_Tot_age_13_17
KPI_PeopleVol_Tot_age_18_29
`

**Review by samshara — APPROVED** [review]

**Review by samshara — APPROVED** [review]

**Pull request merged**